src/test/java/subway/SectionAcceptanceTest.java

```diff
@@ -0,0 +1,163 @@
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.HttpStatus;
+import org.springframework.test.annotation.DirtiesContext;
+import subway.exception.InvalidInputException;
+import subway.line.LineResponse;
+
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThatThrownBy;
+import static org.assertj.core.api.InstanceOfAssertFactories.LIST;
+
+
+@DisplayName("지하철노선 구간 관련 기능")
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.DEFINED_PORT)
+@DirtiesContext(classMode = DirtiesContext.ClassMode.BEFORE_EACH_TEST_METHOD)
+public class SectionAcceptanceTest {
+
+    @BeforeEach
+    void setup() {
+        StationFactory.createStation("banghwa");
+        StationFactory.createStation("gangdong");
+        StationFactory.createStation("macheon");
+        LineFactory.createLine("five", 1L, 2L);
+    }
+
+    /**
+     * Given 지하철노선에 등록되어 있는 하행종점역을 상행역으로 하는 구간을 노선에 추가하고
+     * When 지하철노선을 조회하면
+     * Then 순서대로 방화역 - 강동역 - 마천역이 조회된다
+     */
+    @DisplayName("지하철노선 구간 추가")
+    @Test
+    void addSectionSuccess() {
+        // given
+        ExtractableResponse<Response> response = SectionFactory.createSection(1L, 2L, 3L);
+
+        assertThat(response.statusCode()).isEqualTo(HttpStatus.CREATED.value());
+
+        // when
+        response = LineFactory.getLines();
+
+        // then
+        List<String> stationNames = response.jsonPath().getList("[0].stations.name", String.class);
+        assertThat(stationNames).containsExactly("banghwa", "gangdong", "macheon");
+
+    }
+
+    /**
+     * Given 지하철노선에 등록되어 있는 하행종점역을 상행역으로 하지 않는 구간을 노선에 추가하려고 한다
+     * When 이러한 구간을 노선에 추가하려고 하면
+     * Then 이 요청은 실패해야 한다
+     */
+    @DisplayName("지하철노선 구간 추가 실패")
+    @Test
+    void addSectionFailWhenUpstationNotDownstationofLine() {
+        // given
+        Long wrongUpstationId = 3L;
+        Long downstationId = 2L;
+
+        // when
+        ExtractableResponse<Response> response = SectionFactory.createSection(1L, wrongUpstationId, downstationId);
+
+        // then
+        assertThat(response.statusCode()).isEqualTo(HttpStatus.BAD_REQUEST.value());
+        assertThat(response.body().asString()).contains("해당 노선의 하행 종점역과 새로운 구간의 상행역이 일치해야 합니다.");
+
+
+        // 아래와 같이 하면 예외가 발생하지 않았다고 에러가 납니다 그래서 위와 같이 썼는데 이렇게 해도 되는 걸까요?
+//        assertThatThrownBy(() -> SectionFactory.createSection(1L, wrongUpstationId, downstationId))
+//                .isInstanceOf(InvalidInputException.class)
+//                .hasMessageContaining("해당 노선의 하행 종점역과 새로운 구간의 상행역이 일치해야 합니다.");
+
+    }
+
+    /**
+     * Given 해당 노선에 등록되어있는 역을 새 구간의 하행역으로 해서 추가하려고 한다
+     * When 이러한 구간을 노선에 추가하려고 하면
+     * Then 이 요청은 실패해야 한다
+     */
+    @DisplayName("지하철노선 구간 추가 실패2")
```

Comment:
@DisplayName에  ("지하철노선 구간 추가 실패2")로 추가해도 좋지만, 조금 더 명확하게 실패 시나리오로 작성해보면 어떨까요?
그렇게 된다면, 어떤 실패 상황을 다루고 있는지 즉시 이해할 수 있고, 테스트 코드의 가독성을 높일 수 있을 것 같습니다.
 
참고부탁드립니다~

```suggestion
    @DisplayName("새로운 구간의 하행역은 이미 노선에 존재하는 역이면 안 됩니다.")
```